api/src/main/java/org/pmiops/workbench/auth/DelegatedUserCredentials.java

```diff
@@ -0,0 +1,164 @@
+  //     Creator role granted on this service account. See
+  //     https://cloud.google.com/iam/docs/creating-short-lived-service-account-credentials for more
+  //     details.
+  private String serviceAccountEmail;
+  // The full G Suite email address of the user for whom an access token will be generated.
+  private String userEmail;
+  // The set of Google OAuth scopes to be requested.
+  private List<String> scopes;
+  // The HttpTransport to be used for making requests to Google's OAuth2 token server. If null,
+  // a default NetHttpTransport instance is used.
+  private HttpTransport httpTransport;
+  // The IAM Credentials API client to be used for fetching a signed JWT from Google. If null,
+  // a default API client will be used.
+  private IamCredentialsClient client;
+
+  public DelegatedUserCredentials(
+      String serviceAccountEmail, String userEmail, List<String> scopes) {
+    super();
+    this.serviceAccountEmail = serviceAccountEmail;
+    this.userEmail = userEmail;
+    this.scopes = scopes;
+
+    if (this.scopes == null) {
+      this.scopes = new ArrayList<>();
+    }
+  }
+
+  @VisibleForTesting
+  public void setIamCredentialsClient(IamCredentialsClient client) {
+    this.client = client;
+  }
+
+  @VisibleForTesting
+  public void setHttpTransport(HttpTransport httpTransport) {
+    this.httpTransport = httpTransport;
+  }
+
+  /**
+   * Creates the set of JWT claims representing a service account `serviceAccountEmail` using
+   * domain-wide delegation of authority to generate an access token on behalf of a G Suite user,
+   * `userEmail`.
+   *
+   * <p>For reference, see the ServiceAccountCredentials.createAssertion method which builds a
+   * similar JWT payload in the context of a JWT being self-signed using a service account's private
+   * key.
+   *
+   * @return
+   */
+  @VisibleForTesting
+  public JsonWebToken.Payload createClaims() {
+    JsonWebToken.Payload payload = new JsonWebToken.Payload();
+    payload.setIssuedAtTimeSeconds(Instant.now().getEpochSecond());
+    payload.setExpirationTimeSeconds(
+        Instant.now().getEpochSecond() + ACCESS_TOKEN_DURATION.getSeconds());
+    payload.setAudience(GoogleOAuthConstants.TOKEN_SERVER_URL);
+    payload.setIssuer(this.serviceAccountEmail);
+    payload.setSubject(this.userEmail);
+    payload.set("scope", String.join(" ", this.scopes));
+    return payload;
+  }
+
+  @Override
+  public AccessToken refreshAccessToken() throws IOException {
+    // The first step is to call the IamCredentials API to generate a signed JWT with the
+    // appropriate claims. This call is authorized with application default credentials (ADCs). The
+    // ADC service account may be different from `serviceAccountEmail` if the ADC account has the
+    // roles/iam.serviceAccountTokenCreator role on the `serviceAccountEmail` account.
+    JsonWebToken.Payload payload = createClaims();
+
+    if (client == null) {
```

Comment:
The root issue for me here is having a nullable field to begin with. Instead of accessing the private `client` directly, you could have a `private Optional< IamCredentialsClient > get IamCredentialsClient()`. so it's just `getClient().orElseGet(...)` here. Or even do the default inside the accessor.

I like the closely packed style because someone maintaining the code later will be less likely to just stick something in one of the branches without updating all the other places.